### PR TITLE
fix(generator tool): filter tsp-client diagnostics log

### PR DESCRIPTION
**Pull request by Alancere**
fix: https://github.com/Azure/azure-rest-api-specs/pull/29330/checks?check_run_id=28435203799

### CODE REVIEW

File: eng/tools/generator/cmd/v2/common/cmdProcessor.go [hunk 1]
```diff
@@ -224,12 +223,19 @@ func ExecuteTspClient(path string, args ...string) error {
...
+				if len(strings.TrimSpace(line)) == 0 {
+					continue
+				}
 				if !strings.Contains(line, "npm notice") {
 					newErrInfo = append(newErrInfo, line)
 				}
 			}
 
-			return fmt.Errorf("failed to execute `tsp-client %s`\n%s", strings.Join(args, " "), strings.Join(newErrInfo, "\n"))
+			if len(newErrInfo) > 0 {
```
Comment (raych1) [inline eng/tools/generator/cmd/v2/common/cmdProcessor.go:234]: I don't see the diagnostic logs being filtered out.

File: eng/pipelines/mgmt-auto-release.yml [hunk 2]
```diff
@@ -27,7 +27,7 @@ extends:
               - task: Npm@1
                 inputs:
                   command: 'custom'
-                  customCommand: 'install -g @azure-tools/typespec-client-generator-cli@latest'
+                  customCommand: 'install -g @azure-tools/typespec-client-generator-cli@v0.10.0'
```
Comment (raych1) [inline eng/pipelines/mgmt-auto-release.yml:30]: Can you change the other installation place either?
https://github.com/Azure/azure-sdk-for-go/blob/bc736674c503a9d5bc11745c33cddda30ceda2f8/eng/scripts/automation_init.sh#L62

File: eng/tools/generator/cmd/v2/common/cmdProcessor.go [hunk 3]
```diff
@@ -265,3 +295,60 @@ func ExecuteTypeSpecGenerate(ctx *GenerateContext, emitOptions string, tspClient
...
+			if start != -1 {
+				diagnostics = append(diagnostics, diagnostic{kind: kind, start: start, end: i - 1})
+				start = i
+			} else {
+				start = i
+			}
+			kind = "error"
+		}
+
+		if i == len(lines)-1 {
```
Comment (raych1) [inline eng/tools/generator/cmd/v2/common/cmdProcessor.go:332]: how about `if i == len(lines)-1 && start != -1`

File: eng/tools/generator/cmd/v2/common/cmdProcessor.go [hunk 4]
```diff
@@ -210,29 +213,56 @@ func ExecuteTspClient(path string, args ...string) error {
...
 					continue
 				}
 				if !strings.Contains(line, "npm notice") {
-					newErrInfo = append(newErrInfo, line)
+					newErrMsgs = append(newErrMsgs, line)
+				}
+			}
+
+			// filter diagnostic errors
+			if len(newErrMsgs) == 1 &&
```
Comment (raych1) [inline eng/tools/generator/cmd/v2/common/cmdProcessor.go:245]: Should we check `if len(newErrMsgs) > 1 && ..`?

File: eng/tools/generator/cmd/v2/common/cmdProcessor.go [hunk 5]
```diff
@@ -210,29 +213,56 @@ func ExecuteTspClient(path string, args ...string) error {
...
+				newErrMsgs[0] == "Diagnostics were reported during compilation. Use the `--debug` flag to see the diagnostic output." {
+				newErrMsgs = FilterErrorDiagnostics(strings.Split(stdoutBuffer.String(), "\n"))
+
+				temp := make([]string, 0)
+				for _, line := range newErrMsgs {
+					line := strings.TrimSpace(line)
+					if line == "" {
```
Comment (raych1) [inline eng/tools/generator/cmd/v2/common/cmdProcessor.go:252]: Can we combine line252 and line255?

File: eng/tools/generator/cmd/v2/common/cmdProcessor.go (hunk 4 above)
Comment (Alancere) [reply eng/tools/generator/cmd/v2/common/cmdProcessor.go]: >= 1

File: eng/tools/generator/cmd/v2/common/cmdProcessor.go [hunk 6]
```diff
@@ -259,3 +297,60 @@ func ExecuteTypeSpecGenerate(ctx *GenerateContext, emitOptions string, tspClient
...
+		} else if strings.Contains(line, "error ") {
+			if start != -1 {
+				diagnostics = append(diagnostics, diagnostic{kind: kind, start: start, end: i - 1})
+				start = i
+			} else {
+				start = i
+			}
+			kind = "error"
+		}
+
```
Comment (raych1) [inline eng/tools/generator/cmd/v2/common/cmdProcessor.go:333]: How about move `start = i` to this place for simplification?

**Review by raych1 — APPROVED** [review]

**Pull request merged**